src/zenml/integrations/huggingface/model_registries/huggingface_model_registry.py

```diff
@@ -0,0 +1,668 @@
+
+        Returns:
+            A list of model versions.
+        """
+        models = self.list_models(name=name, metadata={})
+
+        fetched_versions = []
+        for model in models:
+            for discussion in self.api.get_repo_discussions(
+                repo_id=self._repo_id(model.name), repo_type=self.REPO_TYPE
+            ):
+                if (
+                    discussion.is_pull_request
+                    and discussion.title == "New model version from ZenML"
+                ):
+                    if discussion.status == "open":
+                        pr_ref = f"refs/pr/{discussion.num}"
+                        (
+                            description,
+                            found_metadata,
+                            _,
+                        ) = self._get_model_card_content(model.name, pr_ref)
+                        metadata_filter = True
+                        if metadata is not None:
+                            for k, v in metadata.dict().items():
+                                metadata_filter &= (
+                                    k in found_metadata
+                                    and found_metadata[k] == v
+                                )
+                                if not metadata_filter:
+                                    break
+                        if metadata_filter:
+                            fetched_versions.append(
+                                ModelVersion(
+                                    version=pr_ref,
+                                    model_source_uri=discussion.url,
+                                    model_format="hugginface",
+                                    registered_model=model,
+                                    description=description,
+                                    metadata=found_metadata,
+                                )
+                            )
+        return fetched_versions
+
+    def get_model_version(self, name: str, version: str) -> ModelVersion:
+        """Gets a model version for a registered model.
+
+        Args:
+            name: The name of the registered model.
+            version: The version of the model version to get.
+
+        Returns:
+            The model version.
+        """
+        pr_num = version.split("/")[-1]
+        description, metadata, _ = self._get_model_card_content(
+            name=name, revision=version
+        )
+        return ModelVersion(
+            version=version,
+            model_source_uri=f"https://huggingface.co/{self._repo_id(name)}/discussions/{pr_num}",
+            model_format="hugginface",
+            registered_model=self.get_model(name),
+            description=description,
+            metadata=metadata,
+        )
+
+    def load_model_version(
+        self,
+        name: str,
+        version: str,
+        path: str,
+        **kwargs: Any,
+    ) -> None:
+        """Loads a model version from the model registry.
```

Comment:
Something feels off about this. It doesn't return anything, and what it does download it does to a local path / outside the artifact store etc? Feels like a recipe for silent failure etc. Also are we enabling flows that are ultimately non-reproducible?